lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -187,6 +223,67 @@ export class BrowserCacheManager extends CacheManager {
         this.logger.trace("BrowserCacheManager.setAccount called");
         const key = account.generateAccountKey();
         this.setItem(key, JSON.stringify(account));
+        this.addAccountKeyToMap(key);
+    }
+
+    /**
+     * Returns the array of account keys currently cached
+     * @returns 
+     */
+    getAccountKeys(): Array<string> {
+        this.logger.trace("BrowserCacheManager.getAccountKeys called");
+        const accountKeys = this.getItem(StaticCacheKeys.ACCOUNT_KEYS);
+        if (accountKeys) {
+            return JSON.parse(accountKeys);
+        }
+
+        this.logger.verbose("BrowserCacheManager.getAccountKeys - No account keys found");
+        return [];
+    }
+
+    /**
+     * Add a new account to the key map
+     * @param key 
+     */
+    addAccountKeyToMap(key: string): void {
+        this.logger.trace("BrowserCacheManager.addAccountKeyToMap called");
+        this.logger.tracePii(`BrowserCacheManager.addAccountKeyToMap called with key: ${key}`);
+        const accountKeys = this.getAccountKeys();
+        if (accountKeys.indexOf(key) === -1) {
+            // Only add key if it does not already exist in the map
+            accountKeys.push(key);
+            this.setItem(StaticCacheKeys.ACCOUNT_KEYS, JSON.stringify(accountKeys));
```

Comment:
Does this hold if the number of accounts exceed a given limit? `localStorage` or `sessionStorage` entries do have a size limit I assume?